src/iterator.rs

```diff
@@ -13,76 +13,129 @@ use std::iter;
 /// Iterator over a flat-tree.
 pub struct Iterator {
   index: usize,
+  offset: usize,
+  factor: usize,
 }
 
 impl Iterator {
   /// Create a new iterator.
   pub fn new(index: usize) -> Self {
-    Self { index }
+    let mut instance = Self {
+      index: 0,
+      offset: 0,
+      factor: 0,
+    };
+
+    if index != 0 {
+      instance.seek(index);
+    }
+
+    instance
+  }
+
+  /// Get the current index.
+  #[inline]
+  pub fn index(&self) -> usize {
+    self.index
   }
 
   /// Seek to a position in the iterator.
   pub fn seek(&mut self, index: usize) {
     self.index = index;
+    if (self.index & 1) > 0 {
```

Comment:
Can we pull in `is_odd()`/`is_even()` from `src/lib.rs`? If so, this could be `if is_odd(self.index) {}`

Reply:
for sure!